inc/namespace.php

```diff
@@ -31,3 +33,42 @@ function on_plugins_loaded() {
 		require_once ROOT_DIR . '/vendor/johnbillion/query-monitor/query-monitor.php';
 	}
 }
+
+function qm_file_path_map( $map ) : array {
+	// Chassis and Local Server
+	if ( file_exists( '/etc/chassis-constants' ) ) {
+		$json_string = file_get_contents( '/etc/chassis-constants' );
+		$data = json_decode( $json_string, true );
+		if ( ! empty( $data['synced_folders']['/chassis'] ) ) {
+			$folder_path = $data['synced_folders']['/chassis'] . '/content/themes/base';
+			$map['/chassis/'] = $data['synced_folders']['/chassis'] . '/';
+		}
+	}
+
+	return $map;
+}
+
+function qm_file_link_format( $format ) : string {
+	$editor = 'phpstorm';
+	if ( defined( 'QM_LOCAL_EDITOR' ) ) {
+		$editor = QM_LOCAL_EDITOR;
+	}
+	return qm_file_link_editor_format( $format, $editor );
+}
+
+function qm_file_link_editor_format( $format, $editor = null ) : string {
+	switch ( $editor ) {
+		case 'phpstorm':
+			return 'phpstorm://open?file=%f&line=%l';
+		case 'vscode':
+			return 'vscode://file/%f:%l';
+		case 'atom':
+			return 'atom://open/?url=file://%f&line=%l';
+		case 'sublime':
+			return 'subl://open/?url=file://%f&line=%l';
+		case 'netbeans':
+			return 'nbopen://%f:%l';
+		default:
+			return $format;
+	}
+}
```

Comment:
@johnbillion Thoughts on including something like this in QM by default?

Reply:
Yeah sounds good! PR welcome.